### PR TITLE
[DO NOT MERGE] CI Test

### PR DESCRIPTION
**Pull request by joecummings**

### CODE REVIEW

**Review by ebsmothers — COMMENTED** [review]
very nice

File: torchtune/dev/rl/workers/trainers/training.py [hunk 1]
```diff
@@ -372,19 +301,9 @@ def _setup_model(
 
     def _setup_optimizer(
         self, cfg_optimizer: DictConfig, opt_state_dict=None
-    ) -> Optional[Optimizer]:
+    ) -> Optimizer:
```
[the remark this answers is not in the document]
Comment (joecummings) [reply torchtune/dev/rl/workers/trainers/training.py:304]: Yep, if we want to get this out today.

File: recipes/configs/dev/qwen3B_async_grpo.yaml [hunk 2]
```diff
@@ -10,144 +10,103 @@
...
 
-# Tokenizer
+# Model architecture (Qwen2.5-3B)
+model:
+  _component_: torchtune.models.qwen2_5.qwen2_5_3b
 tokenizer:
   _component_: torchtune.models.qwen2_5.qwen2_5_tokenizer
-  path: /tmp/Qwen2.5-3B/vocab.json
-  merges_file: /tmp/Qwen2.5-3B/merges.txt
+  path: ${base_model_path}/vocab.json
```
Comment (Darktex) [inline recipes/configs/dev/qwen3B_async_grpo.yaml:27]: 👏

File: recipes/configs/dev/qwen3B_async_grpo.yaml [hunk 3]
```diff
@@ -10,144 +10,103 @@
...
+    _component_: torchtune.training.FullModelHFCheckpointer
+    checkpoint_dir: ${base_model_path}
+    checkpoint_files: [
+      model-00001-of-00002.safetensors,
+      model-00002-of-00002.safetensors,
+    ]
+    model_type: QWEN2
+
+# All inference args
+inference:
```
Comment (Darktex) [inline recipes/configs/dev/qwen3B_async_grpo.yaml:87]: Nit: maybe we should put inference at the top? I think conceptually that's how the recipe flows: generation, postprocessing, training, weight update

File: torchtune/config/_parse.py [hunk 4]
```diff
@@ -96,6 +95,6 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
         yaml_args, cli_args = parser.parse_known_args()
         conf = _merge_yaml_and_cli_args(yaml_args, cli_args)
 
-        sys.exit(recipe_main(conf))
+        recipe_main(conf)
```
Comment (Darktex) [inline torchtune/config/_parse.py:98]: This is a change to all of Tune. Why is this necessary?

File: torchtune/dev/grpo/loss.py [hunk 5]
```diff
@@ -354,6 +354,13 @@ def forward(
         """
         Compute GRPO loss over chunked or full logits.
 
+        Args:
```
Comment (Darktex) [inline torchtune/dev/grpo/loss.py:357]: 👏

File: torchtune/dev/rl/__init__.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```
Comment (Darktex) [inline torchtune/dev/rl/__init__.py:1]: Thanks for doing this

File: torchtune/dev/rl/workers/datacollectors/sync.py [hunk 7]
```diff
@@ -63,19 +62,18 @@ def __init__(
...
-        print(f"{self._is_collector_zero=}")
-
-        self.tp_size = self.cfg.inference.tp_size
+        self.tp_size = self.cfg.inference.tensor_parallel_dim
         self.batch_size = self.cfg.inference.batch_size
         self._sequence_counter = 0  # Used to assign unique sequence IDs to each sample
 
         self.inference_server = LLM(
-            model="Qwen/Qwen2.5-3B",
+            model=llm,
```
Comment (Darktex) [inline torchtune/dev/rl/workers/datacollectors/sync.py:70]: 👏

File: torchtune/dev/rl/workers/datacollectors/sync.py [hunk 8]
```diff
@@ -63,19 +62,18 @@ def __init__(
...
             enforce_eager=True,
             enable_chunked_prefill=True,
             dtype="bfloat16",
             worker_cls=VLLMWorkerWrapper,
             tensor_parallel_size=self.tp_size,
+            disable_custom_all_reduce=True,
```
Comment (Darktex) [inline torchtune/dev/rl/workers/datacollectors/sync.py:76]: What is this?

File: torchtune/dev/rl/workers/datacollectors/sync.py [hunk 9]
```diff
@@ -239,7 +234,7 @@ def run(self):
         i = 0
         while True:
             self.rollout(i)
-            if i % self.cfg.inference.steps_before_sync == 0:
+            if i % self.cfg.inference.steps_before_weight_sync == 0:
```
Comment (Darktex) [inline torchtune/dev/rl/workers/datacollectors/sync.py:237]: Good rename, me like

File: torchtune/dev/rl/workers/postprocessing.py [hunk 10]
```diff
@@ -31,13 +31,12 @@ def __init__(self, *args, **kwargs):
         self.cfg = kwargs.pop("cfg")
         self.rollout_queue = kwargs.pop("rollout_queue")
         self.replay_buffer = kwargs.pop("replay_buffer")
-        self._device = utils.get_device(device=self.cfg.postprocessing.device_type)
+        device_type = "cuda"
```
Comment (Darktex) [inline torchtune/dev/rl/workers/postprocessing.py:34]: Why hardcode instead of reading it from cfg?

File: torchtune/modules/transformer.py [hunk 11]
```diff
@@ -664,7 +667,7 @@ def forward(
 
     def unembed(self, h, output_mask=None):
         if output_mask is not None:
-            bsz, seq_len, dim = h.shape
+            bsz, _, dim = h.shape
```
Comment (Darktex) [inline torchtune/modules/transformer.py:670]: Does this belong to this PR?

**Review by Darktex — CHANGES_REQUESTED** [review]
There are a few things that seem extraneous to this PR, maybe this is a stack?

File: torchtune/config/_parse.py (hunk 4 above)
Comment (joecummings) [reply torchtune/config/_parse.py:98]: Ahh was debugging - this can go back in.

File: torchtune/dev/rl/workers/postprocessing.py (hunk 10 above)
Comment (joecummings) [reply torchtune/dev/rl/workers/postprocessing.py:34]: We actually hard coded it in many different places and don't have time to test on different devices so this just explicitly sets it to "cuda"

File: torchtune/modules/transformer.py (hunk 11 above)
Comment (joecummings) [reply torchtune/modules/transformer.py:670]: Linting :)
